tests/scenario/test_charm.py

```diff
@@ -0,0 +1,601 @@
+    peer_relation = scenario.PeerRelation(
+        "takahe-peer", "takahe-peer", local_app_data={"secret-id": secret.id}
+    )
+    database_relation = scenario.Relation(
+        "database", "postgresql_client", remote_app_data={"endpoints": endpoint}
+    )
+    # SCENARIO-NOTE: It would be nice if the charm test didn't need to know the
+    # layout, since the lib is hiding all of that - but maybe that's really an
+    # issue with how the lib hides everything.
+    db_secret = scenario.Secret(
+        "1235",
+        {0: {"username": user, "password": password}},
+        label=f"database.{database_relation.relation_id}.user.secret",
+    )
+    state = scenario.State(
+        leader=True,
+        relations=[peer_relation, database_relation],
+        containers=[background_container],
+        storage=[storage],
+        secrets=[secret, db_secret],
+        config={"main-domain": domain, "media-uri": media},
+    )
+
+    ## Act.
+    out = ctx.run(background_container.pebble_ready_event, state)
+
+    ## Assert.
+    cmd = "python3 manage.py runstator"
+    assert_running_state(
+        ctx,
+        out,
+        out.containers[0],
+        user,
+        password,
+        endpoint,
+        secret.contents[1]["takahe-secret-key"],
+        domain,
+        media,
+        "Takahē background service",
+        cmd,
+        background_container.name,
+    )
+    assert out.workload_version == version
+
+
+def assert_running_state(
+    ctx,
+    out,
+    container,
+    db_user,
+    db_password,
+    db_host,
+    secret_key,
+    domain,
+    media,
+    summary,
+    cmd,
+    service_name,
+):
+    takahē_env = {
+        "TAKAHE_DATABASE_SERVER": f"postgresql://{db_user}:{db_password}@{db_host}/takahē?connect_timeout=10",
+        "TAKAHE_SECRET_KEY": secret_key,
+        "TAKAHE_MEDIA_BACKEND": media,
+        "TAKAHE_MAIN_DOMAIN": domain,
+        "TAKAHE_EMAIL_FROM": f"takahē@{domain}",
+        "TAKAHE_AUTO_ADMIN_EMAIL": f"takahē@{domain}",
+        "TAKAKE_USE_PROXY_HEADERS": "True",
+        "TAKAHE_MEDIA_ROOT": str(out.storage[0].get_filesystem(ctx)),
+        "TAKAHE_MEDIA_URL": "https://example.com/",
+    }
+    # SCENARIO-NOTE: it would be nice if there was a cleaner way to create the
+    # plan object.
+    raw_plan = yaml.safe_dump(
+        {
+            "summary": summary,
+            "description": "Service for Takahē ActivityPub instance",
+            "services": {
+                service_name: {
+                    "override": "replace",
+                    "summary": summary,
+                    "command": cmd,
+                    "startup": "enabled",
+                    "environment": takahē_env,
+                },
+            },
+        }
+    )
+    expected_plan = ops.pebble.Plan(raw_plan)
+    # SCENARIO-NOTE: maybe actually an ops note: it would be nice if a Plan had
+    # an __eq__ and basically did this.
+    assert container.plan.to_dict() == expected_plan.to_dict()
```

Comment:
@benhoyt Another one that's more ops/Pebble than Scenario: is there any reason not to have an `__eq__` on `Plan`? It seems like this should be a common action when writing tests.

Reply:
Seems reasonable, yeah, as well as adding a `str | PlanDict` option to the `__init__` method. In general I'm cautious about operator overloading, but it seems pretty reasonable to have an `__eq__` here.

Reply:
I feel like same-type `__eq__` is generally reasonable. Don't worry, I'm not going to suggest `__add__` to combine `Layer`s or anything :laughing:.